vault/router.go

```diff
@@ -174,6 +181,32 @@ func (r *Router) MatchingMount(path string) string {
 	return mount
 }
 
+//MatchingPrefix returns a mount prefix that a path may be a part of
+func (r *Router) MatchingPrefix(path string) string {
+	var existing string = ""
+	fn := func(existing_path string, _v interface{}) bool {
+		fmt.Println(fmt.Sprintf("comparing %v %v", path, existing_path))
```

Comment:
Which is why it's embarrassing 👼 

New commit incorporates the feedback and I added a little blurb to the docs about mount conflicts.